addons/purchase/report/purchase_report.py

```diff
@@ -75,19 +78,22 @@ def _select(self):
                     t.categ_id as category_id,
                     s.currency_id,
                     t.uom_id as product_uom,
-                    sum(l.product_qty/u.factor*u2.factor) as unit_quantity,
                     extract(epoch from age(s.date_approve,s.date_order))/(24*60*60)::decimal(16,2) as delay,
                     extract(epoch from age(l.date_planned,s.date_order))/(24*60*60)::decimal(16,2) as delay_pass,
                     count(*) as nbr_lines,
-                    sum(l.price_unit / COALESCE(NULLIF(cr.rate, 0), 1.0) * l.product_qty)::decimal(16,2) as price_total,
-                    avg(100.0 * (l.price_unit / COALESCE(NULLIF(cr.rate, 0),1.0) * l.product_qty) / NULLIF(ip.value_float*l.product_qty/u.factor*u2.factor, 0.0))::decimal(16,2) as negociation,
-                    sum(ip.value_float*l.product_qty/u.factor*u2.factor)::decimal(16,2) as price_standard,
-                    (sum(l.product_qty * l.price_unit / COALESCE(NULLIF(cr.rate, 0), 1.0))/NULLIF(sum(l.product_qty/u.factor*u2.factor),0.0))::decimal(16,2) as price_average,
+                    sum(l.price_total / CASE COALESCE(s.currency_rate, 1.0) WHEN 0 THEN 1.0 ELSE s.currency_rate END)::decimal(16,2) as price_total,
+                    (sum(l.product_qty * l.price_unit / CASE COALESCE(s.currency_rate, 1.0) WHEN 0 THEN 1.0 ELSE s.currency_rate END)/NULLIF(sum(l.product_qty/u.factor*u2.factor),0.0))::decimal(16,2) as price_average,
                     partner.country_id as country_id,
                     partner.commercial_partner_id as commercial_partner_id,
                     analytic_account.id as account_analytic_id,
                     sum(p.weight * l.product_qty/u.factor*u2.factor) as weight,
-                    sum(p.volume * l.product_qty/u.factor*u2.factor) as volume
+                    sum(p.volume * l.product_qty/u.factor*u2.factor) as volume,
```

Comment:
It would be a great idea to rewrite this query and use acronym that mean something, it would be more understandable.
s -> po
u -> line_uom
u2 -> product_uom